engine/src/main/scala/cromwell/engine/backend/io/filesystem/gcs/GcsFileSystemProvider.scala

```diff
@@ -82,26 +93,10 @@ class GcsFileSystemProvider private (storageClient: Try[Storage], executionConte
 
   private def exists(path: Path) = path match {
     case gcsPath: NioGcsPath =>
-      val getObject = client.objects.get(gcsPath.bucket, gcsPath.objectName)
-
-      def checkExists = {
-        Try(getObject.execute) recoverWith {
-          case ex: GoogleJsonResponseException if ex.getStatusCode == 404 =>
-            if (gcsPath.isDirectory) Success(gcsPath) else Failure(ex)
-        }
-      }
-
-      def tryExists(retries: Int): Boolean = checkExists match {
-        case Success(_) => true
-        case Failure(ex: GoogleJsonResponseException) if ex.getStatusCode == 404 && retries > 0 =>
-          // FIXME remove this sleep
-          Thread.sleep(retryInterval.toMillis.toInt)
-          tryExists(retries - 1)
-        case Failure(ex: GoogleJsonResponseException) if ex.getStatusCode == 404 => false
-        case Failure(ex) => throw ex
-      }
-
-      if (!tryExists(retryCount)) throw new FileNotFoundException(path.toString)
+      Try(withRetry(client.objects.get(gcsPath.bucket, gcsPath.objectName).execute)) recover {
+        case ex: GoogleJsonResponseException
+          if ex.getStatusCode == 404 && !gcsPath.isDirectory => throw new FileNotFoundException(path.toString)
```

Comment:
So I realize now this won't work, because it will still throw an exception if gcsPath is a directory on the `get` call below. Since gcs directories never really "exist", `execute` will return 404, although in order for the Nio magic to work we can't throw a FileNotFound exception every time we manipulate directories.
This should be 
```scala
if ex.getStatusCode == 404 => 
   if (!gcsPath.isDirectory) throw new FileNotFoundException(path.toString)
```
I can't fix it right now but I'll try to do it later tonight or tomorrow.
If for some reason I can't and someone take back this PR, don't forget to change this before merging :)